fh-messaging/lib/fh_metrics_man.js

```diff
@@ -197,24 +197,67 @@ MetricsManager.prototype.deleteMetricsData = function(topic, date, callback) {
   // if multiple message managers pointed at the same config are running at the
   // same time
   var msgConfig = JSON.parse(JSON.stringify(self.mConfig));
-  var collectionName = topic + "_" + helpers.toYYYYMMDD(date);
-
   messageMan = new fhmsg.Messaging(msgConfig, self.mLogger);
   self.messageMans.push(messageMan);
+
   messageMan.database.on('tearUp', function() {
-    messageMan.database.db.dropCollection(collectionName, callback);
-    //remove all the data that is older than the x number of days from the intermediate tables
-    async.each(['transactionsdesttemp', 'transactionsgeotemp'], function(tempCol, cb) {
-      var daysToCalculate = self.mConfig.agenda.jobs.metrics_rollup_job.daysToKeep || 31;
-      var beforeDate = new Date(date.getFullYear(), date.getMonth(), date.getDate() - daysToCalculate);
-      var query = {"_id.ts":{$lt: beforeDate.getTime()}};
-      messageMan.database.db.collection(tempCol, function(err, collection) {
+    // Pruning for fhact
+    var daysToCalculate = self.mConfig.agenda.jobs.metrics_rollup_job.daysToKeep || 31;
+    var pruning_date = new Date().getTime() - ( daysToCalculate * 24 * 60 * 60 * 1000 );
+    var prune_collection_names = [];
+
+    // Retrieve the list of collections from the database
+    messageMan.database.db.listCollections({}).toArray( function(err, items) {
+      if (err) {
+        self.mLogger.error("listCollections error: " + err);
+        callback(err);
+      }
+
+      // Generate list of prunable collections
+      for (var i=0; i < items.length; i++) {
+        if (items[i].name.includes(topic +"_")) {
```

Comment:
Would be worth checking items[i].name is defined before calling includes(). This if could be switched to `if (!items[i].name || !items[i].name.includes(topic +"_")) { continue; }` and the rest of the code won't have to be nested in the if statement.